scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+        fi
+        to_wait=$(( $to_wait - 1 ))
+        sleep 1
+    done
+
+    wsrep_log_info "Joiner setup."
+    setup_clone_plugin "recipient"
+
+    wsrep_log_info "Waiting for clone recipient daemon to finish"
+
+    set +e
+    monitor_sst_progress "${MYSQL_ACLIENT} -NB -e 'SELECT * FROM performance_schema.clone_progress;'" ${WSREP_SST_IDLE_TIMEOUT} ${CLONE_INSTANCE_PID} ${PARENT_PID}
+    status=$?
+    set -e
+
+    if [ $status -eq 1 ]; then
+        # parent process died, we need to exit as well
+        wsrep_log_error "Parent mysqld process (PID:$WSREP_SST_OPT_PARENT) terminated unexpectedly."
+        exit $EPIPE
+    fi
+    if [ $status -eq 2 ]; then
+        # Stale SST, no progress
+        wsrep_log_error "******************* FATAL ERROR ********************** "
+        wsrep_log_error "Killing SST ($CLONE_INSTANCE_PID) with SIGKILL after stalling for ${WSREP_SST_IDLE_TIMEOUT} seconds."
+        wsrep_log_error "Within the last ${WSREP_SST_IDLE_TIMEOUT} seconds (defined by the sst-idle-timeout variable),"
+        wsrep_log_error "the SST process on the joiner (this node) has not received any data from the donor."
+        wsrep_log_error "This error could be caused by broken network connectivity between"
+        wsrep_log_error "the donor and the joiner (this node)."
+        wsrep_log_error "Check the network connection and restart the joiner node."
+        wsrep_log_error "Line $LINENO"
+        wsrep_log_error "****************************************************** "
+        exit $ETIMEDOUT
+    fi
+
+    wsrep_log_info "clone recepient daemon finished"
+
+    CLONE_INSTANCE_PID=""
+
+    # Execute first restart for
+    wsrep_log_info "FIRST RESTART to fix the dictionary"
+    wsrep_log_info "Performing data recovery"
+    wsrep_log_debug "-> RECOVERY COMMAND LINE: $CLONE_ENV $CLONE_BINARY $DEFAULT_OPTIONS --wsrep_provider=none"
+
+
+    # Remove created clone user and SST pxc user
+    wsrep_log_debug "-> CLEAN OR NOT? $CLEANUP_CLONE_PLUGIN"
+    CLEANUP_CLONE_PLUGIN_SQL=""
+    if [ "${CLEANUP_CLONE_PLUGIN}" == "yes" ]; then
+            CLEANUP_CLONE_PLUGIN_SQL="UNINSTALL PLUGIN CLONE;"
+    fi
+cat << EOF > "$CLONE_SQL"
+SET SESSION sql_log_bin=OFF;
+DROP USER IF EXISTS $CLONE_USER;
+DROP USER IF EXISTS $CLONE_USER@'localhost';
+DROP USER IF EXISTS 'mysql.pxc.sst.user'@'localhost';
+$CLEANUP_CLONE_PLUGIN_SQL
+SHUTDOWN;
+EOF
+    
+    eval $CLONE_ENV $CLONE_BINARY $DEFAULT_OPTIONS --wsrep_provider=none --init_file="$CLONE_SQL"
+
+    wsrep_log_info "SHUTDOWN done"
+    wsrep_log_info "Second restart for recovery position"
+    wsrep_log_debug "-> Second restart: $CLONE_BINARY $DEFAULT_OPTIONS $SKIP_NETWORKING_OPTIONS --wsrep_recover"
+    eval $CLONE_ENV $CLONE_BINARY $DEFAULT_OPTIONS $SKIP_NETWORKING_OPTIONS --wsrep_recover >> $CLONE_ERR 2>&1
+
+    RP="$(grep -a '\[WSREP\] Recovered position:' $CLONE_ERR || :)"
+    RP_PURGED=`echo $RP | sed 's/.*WSREP\]\ Recovered\ position://' | sed 's/^[ \t]*//'`
```

Comment:
Error returned by shellcheck

SC2006: Use $(...) notation instead of legacy backticks \`...\`.

Reply:
done